docs/Usage.md

```diff
@@ -0,0 +1,323 @@
+
+2. **Create a BucketClaim**
+
+   A `BucketClaim` requests a new bucket using the above `BucketClass`:
+
+   ```bash
+   cat <<EOF | kubectl apply -f -
+   apiVersion: objectstorage.k8s.io/v1alpha1
+   kind: BucketClaim
+   metadata:
+     name: my-greenfield-bucketclaim
+     namespace: container-object-storage-system
+   spec:
+     bucketClassName: greenfield-bucketclass
+     protocols:
+       - S3
+   EOF
+   ```
+
+   - This will automatically provision a new S3 bucket in the backend using the COSI driver. The COSI driver will use credentials and endpoint mentioned in the secret specified in the `BucketClass` to create the bucket.
+   - The actual bucket name on S3 is typically generated by the driver (e.g., `<bucketclassName>-<UUID>`).
+   - Only `S3` protocol is supported at the moment.
+
+### 1.2 Brownfield: Using an Existing Bucket
+
+Brownfield provisioning allows you to manage an **already-existing** S3 bucket in Kubernetes.
+
+> Note: For brownfield scenario, COSI CRs for Bucket and Access provisioning should be created in the same namespace as COSI driver and controller.
+
+1. **Verify Existing Bucket**
+
+   Ensure the bucket already exists in S3 either through Storage Administrator or by running the following AWS CLI command:
+
+   ```bash
+   aws s3api head-bucket --bucket <EXISTING_BUCKET_NAME> --endpoint-url <S3_ENDPOINT>
+   ```
+
+2. **Create a BucketClass**
+
+   Similar to Greenfield, but you will typically still reference the same secret:
+
+   ```bash
+   cat <<EOF | kubectl apply -f -
+   apiVersion: objectstorage.k8s.io/v1alpha1
+   kind: BucketClass
+   metadata:
+     name: brownfield-bucketclass
+     namespace: container-object-storage-system
+   driverName: cosi.scality.com
+   deletionPolicy: Delete
+   parameters:
+     objectStorageSecretName: s3-secret-for-cosi
+     objectStorageSecretNamespace: default
+   EOF
+   ```
+
+   > **Note**
+   > For Brownfield, existing buckets when imported using the steps below, do not follow the `deletionPolicy` even if it set to `Delete` . All buckets created using this bucket class for greenfield scenario will still respect the `deletionPolicy`
+
+3. **Create the Bucket Instance**
+   This is where we tell Kubernetes about the existing bucket:
+
+   ```bash
+   cat <<EOF | kubectl apply -f -
+   apiVersion: objectstorage.k8s.io/v1alpha1
+   kind: Bucket
+   metadata:
+     name: "<EXISTING_BUCKET_NAME>"
+     namespace: container-object-storage-system
+   spec:
+     bucketClaim: {}
+     driverName: cosi.scality.com
+     bucketClassName: brownfield-bucketclass
+     driverName: cosi.scality.com
+     deletionPolicy: Retain
+     existingBucketID: "<EXISTING_BUCKET_NAME>"
+     parameters:
+       objectStorageSecretName: s3-secret-for-cosi
+       objectStorageSecretNamespace: default
+     protocols:
+       - S3
+   EOF
```

Comment:
remove indent

Reply:
for easy copy paste

Reply:
I would like to keep the indent, as Kubernetes throws parsing errors and rejects improperly formatted YAML.